A21-Service-Config-Error-Handling.md

```diff
@@ -0,0 +1,73 @@
+Service Config Error Handling
+----
+* Author: yashkt
+* Approver: markdroth
+* Status: Draft
+* Implemented in:
+* Last updated: 09-17-2018
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/bAxyse6c2eI
+
+## Abstract
+
+This document proposes a methodology to handle errors in a resolved service
+config.
+
+## Background
+
+The
+[service config](https://github.com/grpc/grpc/blob/master/doc/service_config.md)
+is a mechanism that allows service owners to publish
+parameters that clients could use. If the received service config is malformed
+or has unknown fields or has invalid values, the client could either ignore the
+invalid service config or partially accept the service config while ignoring
+certain parts.
+
+
+### Related Proposals:
+* [A2-Service Config via DNS](A2-service-configs-in-dins.md)
+
+## Proposal
+
+If a resolved service config is badly formatted or contains invalid values for
```

Comment:
There are a ton of failure cases here.  We need to get consensus about what values are errors so that we get consistent behavior.

Reply:
What values are considered invalid would depend on the field itself? For example, one proposal to make it easier to roll out new loadbalancing policies is to convert loadBalancingPolicy into a list, and the client would accept the first policy it recognizes. Only if it does not recognize any of the policies in the list would the value be considered invalid.

Reply:
Also, please continue this discussion at https://groups.google.com/forum/#!topic/grpc-io/bAxyse6c2eI